### DISCUSSION TITLE
Changing JWTs to RS256 to verify JWT inside getSession() anywhere

### DISCUSSION THREAD

**Discussion by gc-ft** [supabase/supabase#28873] [Feature Requests] [answered]
Hi,

right now the supabase JWTs use HS256 for encryption. This means that in order to verify the JWT fully we need the JWT secret, which in turn means that the best way to verify the JWT right now is to query supabase.auth.getUser() which retrieves the user from the auth backend while checking the JWT is ok. getSession() does (as the discussions around the warnings it can generate in connection with SSR) NOT verify the JWT. Therefore the JWT payload could be modified, without changing the JWT signature and you would never know.

Having to run supabase.auth.getUser() on every server request or possibly more often is however not very cost-effective and takes time.

Is the best solution not to change Supabase's JWT to RS256 or RS512, which would result in a private key to generate the signature but a public key to verify it! This would mean it would be totally ok to release the public key even inside the client facing code, allowing both server and client to use getSession() and very safely verify the JWT signature without a single call to the server.

This to me seems like a very simple change which in turn makes the system much more effective?

**Comment by GaryAustin1** [comment, marked as answer]
There is work underway to do have asymmetric JWTs.  They have said it is a top priority.
https://github.com/supabase/auth-js/issues/873#issuecomment-2092401090

**Reply by gc-ft** [reply]
Ah, great.. then my thinking was not wrong... I remembered I read something somewhere but could not find it this morning...

Hopefully released soon!

**Reply by attex** [reply]
Is there something like a development roadmap? 

Would be good to know whether that's about to released within the next few days/weeks or months.
